src/iterators/idlist_iterator.c

```diff
@@ -23,6 +23,7 @@ static size_t IL_NumEstimated(QueryIterator *base) {
 *  Returns ITERATOR_EOF if at the end */
 static IteratorStatus IL_Read(QueryIterator *base) {
   IdListIterator *it = (IdListIterator *)base;
+  RS_ASSERT(it->offset < it->size);
```

Comment:
This assertion is invalid; With the current implementation, we may have `it->offset >= it->size` and the following check will catch it